### PR TITLE
DRYD-1436: Add related link group

### PR DESCRIPTION
**Pull request by mikejritter**
**What does this do?**
* Add related link group to collectionobject

**Why are we doing this? (with JIRA link)**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1436

This adds related links to the reference information section for collectionobjects.

**How should this be tested? Do these changes have associated tests?**
* Rebuild collectionspace
* Using the related cspace-ui PR: 
  * Create a collectionobject and scroll to the Reference Information section
  * See that the related link group exists and can be filled out
  * Switch to the public browser template and see the same as the above

**Dependencies for merging? Releasing to production?**
None

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter tested locally

### CODE REVIEW

File: tomcat-main/src/main/resources/defaults/base-collectionobject.xml [hunk 1]
```diff
@@ -460,6 +460,11 @@
 			<field id="reference" autocomplete="true" />
 			<field id="referenceNote" />
 		</repeat>
+
+		<repeat id="relatedLinkGroupList/relatedLinkGroup">
+			<field id="relatedURL" />
```
Comment (ray-lee) [inline tomcat-main/src/main/resources/defaults/base-collectionobject.xml:465]: In the UI, this is called `relatedLink`.

**Review by ray-lee — CHANGES_REQUESTED** [review]
There seems to be a discrepancy in field names between application and UI.

File: tomcat-main/src/main/resources/defaults/base-collectionobject.xml [hunk 2]
```diff
@@ -460,6 +460,11 @@
 			<field id="reference" autocomplete="true" />
 			<field id="referenceNote" />
 		</repeat>
+
+		<repeat id="relatedLinkGroupList/relatedLinkGroup">
```
[the remark this answers is not in the document]
Comment (mikejritter) [reply tomcat-main/src/main/resources/defaults/base-collectionobject.xml]: Looks like I never pushed my commit, these should have been `publishedRelatedLink` to be aligned with the ui.

**Review by ray-lee — APPROVED** [review]
Looks good.

**Pull request merged**